### PR TITLE
Add OAP E2E test case

### PR DESCRIPTION
**Pull request by lujiajing1126**
1. Upload a specific commit of OAP server,
2. Copy all necessary E2E files to this repo

### CODE REVIEW

**Comment by wu-sheng** [comment]
Which OAP are you using? A private fork?

**Comment by lujiajing1126** [comment]
> Which OAP are you using? A fork?

The branch I'am working on.

**Comment by kezhenxu94** [comment]
files under `test/e2e-v2/java-test-service/e2e-protocol` should be referenced via github module

**Comment by wu-sheng** [comment]
So, you have to rebuild the package at banyandb side?
If so, please don't merge this, because that commit will disappear in the future.

**Comment by lujiajing1126** [comment]
> files under `test/e2e-v2/java-test-service/e2e-protocol` should be referenced via github module

Thanks for reminding. Fixed.

**Comment by lujiajing1126** [comment]
> So, you have to rebuild the package at banyandb side? If so, please don't merge this, because that commit will disappear in the future.

Currently, the image is built on a local laptop and upload to the container registry, i.e. https://github.com/apache/skywalking-banyandb/pkgs/container/skywalking-banyandb%2Foap

Our intention is that we would like to test on a specific OAP version during the integration of OAP. But we cannot have a working OAP image for testing until the OAP-integration branch is finally merged to the master branch of the main repo, right? Any suggestion then?

**Comment by wu-sheng** [comment]
Please don't do this(pull private image to an official repo). Delete the private one if possible..
If you need an image, your docker hub is free to use, we are OK to link a personal image repo for testing.

**Comment by lujiajing1126** [comment]
> Please don't do this(push a private image to an official repo). Delete the private one if possible.. If you need an image, your docker hub is free to use, we are OK to link a personal image repo for testing.

Removed (via GitHub Rest API). Sure I could upload to the official hub or `quay.io`.

**Comment by wu-sheng** [comment]
Yes, anyplace you like. Just don't merge this PR, until main repo has official image.

File: test/e2e-v2/cases/storage/banyandb/e2e.yaml
Comment (hanahmily) [inline test/e2e-v2/cases/storage/banyandb/e2e.yaml:45]: Where is this file?

File: test/e2e-v2/script/docker-compose/base-compose.yml
Comment (hanahmily) [inline test/e2e-v2/script/docker-compose/base-compose.yml:30]: Do we have to download mysql?

**Review by hanahmily — COMMENTED** [review]
Do we have to host `test/e2e-v2/java-test-service`? How about building several images? @wu-sheng Can we build some private images here?

File: test/e2e-v2/script/docker-compose/base-compose.yml
Comment (wu-sheng) [reply test/e2e-v2/script/docker-compose/base-compose.yml:30]: No, we don't. I think this is a copy-paste issue.

**Comment by wu-sheng** [comment]
> Can we build some private images here?

Officially we don't. At least not for now. But we could add this for the main repo.

**Comment by lujiajing1126** [comment]
@hanahmily E2E passed!

**Comment by wu-sheng** [comment]
> Continuous Integration / Test (pull_request) 

UT seems a little unstable.

File: test/e2e-v2/cases/storage/banyandb/docker-compose.yml
Comment (wu-sheng) [inline test/e2e-v2/cases/storage/banyandb/docker-compose.yml:66]: As we need to pend on @hanahmily 's measure side works, could you work on main repo to support e2e test service images available?

Comment (lujiajing1126) [reply test/e2e-v2/cases/storage/banyandb/docker-compose.yml:66]: Sure. I could submit another PR to publish these images in the main repo.

**Comment by lujiajing1126** [comment]
> > Continuous Integration / Test (pull_request)
> 
> UT seems a little unstable.

Yes. We will investigate this issue.

File: test/e2e-v2/cases/storage/banyandb/docker-compose.yml
Comment (wu-sheng) [reply test/e2e-v2/cases/storage/banyandb/docker-compose.yml:66]: https://github.com/apache/skywalking/pkgs/container/skywalking%2Fe2e-service-* should be a good prefix for your image.

Comment (lujiajing1126) [reply test/e2e-v2/cases/storage/banyandb/docker-compose.yml:66]: > https://github.com/apache/skywalking/pkgs/container/skywalking%2Fe2e-service-* should be a good prefix for your image.

Got it

File: .github/actions/sw-java-agent/action.yml
Comment (wu-sheng) [inline .github/actions/sw-java-agent/action.yml:33]: Need to download from here, https://archive.apache.org/dist/skywalking/. The mirror is only providing latest version, it changes from time to time.

Comment (lujiajing1126) [reply .github/actions/sw-java-agent/action.yml]: Thanks for the information. Fixed

**Comment by lujiajing1126** [comment]
Ready for review again @wu-sheng @hanahmily @kezhenxu94

File: .github/actions/skip/action.yml
Comment (wu-sheng) [inline .github/actions/skip/action.yml:39]: You should update this. I think these are legecy.

File: .licenserc.yaml
Comment (wu-sheng) [inline .licenserc.yaml:81]: I think we don't use submodule in go project?

File: .github/actions/skip/action.yml
Comment (lujiajing1126) [reply .github/actions/skip/action.yml]: Removed

File: .licenserc.yaml
Comment (lujiajing1126) [reply .licenserc.yaml]: Fixed

File: test/e2e-v2/script/docker-compose/base-compose.yml
Comment (wu-sheng) [inline test/e2e-v2/script/docker-compose/base-compose.yml:70]: Please lock the version in case of any breaks in the future(eventually it will), choose share-volume and make sure to use `SW_AGENT_JAVA_COMMIT` to get the locked agent version.

**Review by wu-sheng — DISMISSED** [review]

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**
